### ISSUE TITLE
Convergence error with localized expression coefficients

### ISSUE THREAD

**Issue by birkantunc** [compsygroup/3DI#12]
When estimating localized expressions coefficients (compute_local_exp_coefficients.py), we get many of the following warnings (even after installing the specific version of scikit-learn==1.1.3)

```
/opt/3DI/build/env/lib64/python3.8/site-packages/sklearn/linear_model/_coordinate_descent.py:614: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Duality gap: 873.6964420412301, tolerance: 4.245814455839991
  model = cd_fast.enet_coordinate_descent_gram(
```
`duality gap: 873.6964420412301` seems to be very large. Does this mean estimated values are not reliable?

**Comment by sariyanidi** [comment]
This may have to do with the bug that you discovered. We'll need to look at this

**Comment by sariyanidi** [comment]
When I fixed #13 the duality gap decreased significantly to ~200. But still too large. I'm re-training a basis (with the large sample) and will see if this issue persists.

**Comment by sariyanidi** [comment]
Looks like convergence issues disappear when we re-train a new basis with a larger sample.
